src/client/datascience/data-viewing/data-wrangler/dataWranglerProvider.ts

```diff
@@ -0,0 +1,157 @@
+import { injectable, inject, named } from 'inversify';
+import { noop } from 'lodash';
+import {
+    CustomDocument,
+    Event,
+    WebviewPanel,
+    Uri,
+    CustomDocumentEditEvent,
+    EventEmitter,
+    window,
+    ProgressLocation
+} from 'vscode';
+import { IExtensionSingleActivationService } from '../../../activation/types';
+import { IApplicationShell, IDataWranglerProvider } from '../../../common/application/types';
+import * as uuid from 'uuid/v4';
+import { Identifiers } from '../../constants';
+import { INotebookProvider, IJupyterVariables, IJupyterVariableDataProviderFactory } from '../../types';
+import { DataViewerChecker } from '../../interactive-common/dataViewerChecker';
+import { IConfigurationService } from '../../../common/types';
+import { IDataWranglerFactory } from './types';
+import { DataScience } from '../../../common/utils/localize';
+import { IDataViewerDataProvider } from '../types';
+
+@injectable()
+export class DataWranglerProvider implements IDataWranglerProvider, IExtensionSingleActivationService {
```

Comment:
can you put a comment on the top of each class explaining what it's for? For example it's not clear immediately the difference between DataWranglerProvider and DataWranglerFactory

Reply:
Can you put this same comment on all of the new classes you added?